### PR TITLE
Add physical only cell list API

### PR DESCRIPTION
**Pull request by colinschmidt**
To be used when creating an LVS netlist

### CODE REVIEW

**Comment by colinschmidt** [comment]
I'm currently using this in my favorite technology, and it solves my use case.

File: src/hammer-tech/hammer_tech.py [hunk 1]
```diff
@@ -404,6 +404,20 @@ def dont_use_list(self) -> Optional[List[str]]:
...
+        """
+        Get the list of physical only  cells.
+        :return: List of physical only cells, or None if the technology does not define such a list.
+        """
+        physical_only_cells_list_raw = self.config.physical_only_cells_list  # type: Optional[List[str]]
+        if physical_only_cells_list_raw is None:
+            return None
+        else:
+            # Work around the weird objects implemented by the jsonschema generator.
+            physical_only_cells_list = list(map(lambda x: str(x), list(physical_only_cells_list_raw)))
```
Comment (edwardcwang) [inline src/hammer-tech/hammer_tech.py]:
```suggestion
            physical_only_cells_list = list(map(str, list(physical_only_cells_list_raw)))
```

File: src/hammer-tech/hammer_tech.py [hunk 2]
```diff
@@ -404,6 +404,20 @@ def dont_use_list(self) -> Optional[List[str]]:
             dont_use_list = list(map(lambda x: str(x), list(dont_use_list_raw)))
             return dont_use_list
 
+    @property
+    def physical_only_cells_list(self) -> Optional[List[str]]:
+        """
+        Get the list of physical only  cells.
```
Comment (edwardcwang) [inline src/hammer-tech/hammer_tech.py]:
```suggestion
        Get the list of physical only cells.
```

File: src/hammer-vlsi/defaults.yml [hunk 3]
```diff
@@ -454,6 +454,21 @@ par.inputs:
   # type: bool
   gds_merge: false
 
+  # Physical only cells mode.
+  # type: str
+  # The list of physical only cells are cells which don't have a logic function.
+  # They won't have a spice circuit for instance.
```
Comment (edwardcwang) [inline src/hammer-vlsi/defaults.yml]:
```suggestion
  # They won't have a SPICE circuit for instance.
```

File: src/hammer-vlsi/defaults.yml [hunk 4]
```diff
@@ -454,6 +454,21 @@ par.inputs:
...
+  # This is currently only used for what cells to exclude from the LVS netlist, which
+  # requires spice circuits for all cells.
```
Comment (edwardcwang) [inline src/hammer-vlsi/defaults.yml]:
```suggestion
  # requires SPICE circuits for all cells.
```

File: src/hammer-vlsi/defaults.yml [hunk 5]
```diff
@@ -454,6 +454,21 @@ par.inputs:
   # type: bool
   gds_merge: false
 
+  # Physical only cells mode.
+  # type: str
+  # The list of physical only cells are cells which don't have a logic function.
+  # They won't have a spice circuit for instance.
+  # This is currently only used for what cells to exclude from the LVS netlist, which
```
Comment (edwardcwang) [inline src/hammer-vlsi/defaults.yml]: Does this include filler/decap cells? May be related to #180 

Also, how does it affect LVS? Does it mean that if a physical-only cell causes a short, it won't be detected?

File: src/hammer-tech/hammer_tech.py (hunk 1 above)
Comment (jwright6323) [inline src/hammer-tech/hammer_tech.py]: Getting into the holy war a bit:
```suggestion
            physical_only_cells_list = [str(x) for x in physical_only_cells_list_raw]
```

File: src/hammer-vlsi/defaults.yml (hunk 5 above)
Comment (jwright6323) [reply src/hammer-vlsi/defaults.yml]: Well, filler and decap cells can and should be treated differently here. Filler cells we remove- they have no LVS-able devices in them. Decap cells do. LVS should still detect a short.

File: src/hammer-tech/hammer_tech.py (hunk 1 above)
Comment (edwardcwang) [reply src/hammer-tech/hammer_tech.py]: Just FYI that `map(func, ...)` is faster than its list comprehension equivalent as opposed to `map(lambda x: <custom logic>, ...)` so would be good to favour `map(str, ...)` in this case

**Pull request merged**